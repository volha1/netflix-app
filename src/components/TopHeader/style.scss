@import '../../common/scss/base.scss';

.top-header {
  display: flex;
  justify-content: space-between;
  margin-bottom: 50px;
}

.add-btn {
  width: 177px;
  height: 46px;
  font-weight: 600;
  color: $red;
  background-color: rgba(85, 85, 85, 0.7);

  &:hover {
    background-color: rgba(85, 85, 85, 1);
  }
}
