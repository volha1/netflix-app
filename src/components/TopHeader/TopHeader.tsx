import React, { ReactElement } from 'react';
import './style.scss';

type TopHeaderProps = {
  handleMovieForm: () => void;
};

const TopHeader = ({ handleMovieForm }: TopHeaderProps): ReactElement => {
  return (
    <div className="top-header">
      <p className="logo">
        <span>netflix</span>roulette
      </p>
      <button type="button" className="btn add-btn" onClick={handleMovieForm}>
        + Add movie
      </button>
    </div>
  );
};

export default TopHeader;
