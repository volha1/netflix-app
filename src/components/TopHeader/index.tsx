import React from 'react';
import './style.scss';

const TopHeader = () => {
  return (
    <div className="top-header">
      <p className="logo">
        <span>netflix</span>roulette
      </p>
      <button type="button" className="btn add-btn">
        + Add movie
      </button>
    </div>
  );
};

export default TopHeader;
