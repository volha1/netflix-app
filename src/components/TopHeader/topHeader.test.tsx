import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import TopHeader from './TopHeader';

describe('TopHeader test', () => {
  const onAddMovieForm = jest.fn();

  it('TopHeader snapshot', () => {
    const component = renderer.create(<TopHeader onAddMovieForm={onAddMovieForm} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('onAddMovieForm works', async () => {
    render(<TopHeader onAddMovieForm={onAddMovieForm} />);

    await userEvent.click(screen.getByRole('button'));

    expect(onAddMovieForm).toHaveBeenCalled();
  });
});
