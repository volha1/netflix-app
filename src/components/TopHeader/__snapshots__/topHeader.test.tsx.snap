// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopHeader test TopHeader snapshot 1`] = `
<div
  className="top-header"
>
  <p
    className="logo"
  >
    <span>
      netflix
    </span>
    roulette
  </p>
  <button
    className="btn add-btn"
    onClick={[MockFunction]}
    type="button"
  >
    + Add movie
  </button>
</div>
`;
